ios_system.m

```diff
@@ -51,6 +51,34 @@
 __thread FILE* thread_stderr;
 __thread void* thread_context;
 
+FILE* ios_stdin(void) {
+    return thread_stdin;
+}
+
+FILE* ios_stdout(void) {
+    return thread_stdout;
+}
+
+FILE* ios_stderr(void) {
+    return thread_stderr;
+}
+
+int ios_stdinFd(void) {
+    return thread_stdin ? fileno(thread_stdin) : 0;
+}
+
+int ios_stdoutFd(void) {
+    return thread_stdout ? fileno(thread_stdout) : 0;
```

Comment:
I see your point. But 0 is STDIN_FILENO (or fileno(stdin)), so as a return value, it won't necessarily be treated as an error by the caller. Also, if the caller forgets to check the return value, we will get a crash (because trying to write on stdin will cause a crash). I would prefer to have a behaviour that does not crash, even if it writes everything off-screen.

Reply:
Force pushed without these helpers